### PR TITLE
Upgraded coremltools to version 3.1.

### PR DESCRIPTION
**Pull request by hoytak**
Fixes https://github.com/apple/turicreate/issues/2774.

### CODE REVIEW

**Comment by hoytak** [comment]
@TobyRoseman -- I see what you're saying, but it also makes very little sense to depend on a beta version when the non-beta has been released.

File: src/python/turicreate/toolkits/sound_classifier/sound_classifier.py [hunk 1]
```diff
@@ -806,35 +805,39 @@ def get_custom_model_spec():
...
-            builder.set_input([input_name], [(input_length,)])
-            builder.set_output([self.target], [(self.num_classes,)])
+                input_name = output_name
 
+                if cur_layer['act']:
+                    output_name = next_layer_name()
+                    builder.add_activation("activation"+str(i), 'RELU', input_name, output_name)
+                    input_name = output_name
+
+            builder.add_softmax('softmax', input_name, self.target)
```
Comment (srikris) [inline src/python/turicreate/toolkits/sound_classifier/sound_classifier.py]: The bug was simply a difference in behavior between 3.0b1 and 3.1 where setting class labels before output vs after output overrides the class names.

**Comment by srikris** [comment]
Passes all tests on all platforms.

**Comment by hoytak** [comment]
Looks good @srikris !

File: src/python/turicreate/toolkits/sound_classifier/sound_classifier.py (hunk 1 above)
Comment (TobyRoseman) [reply src/python/turicreate/toolkits/sound_classifier/sound_classifier.py]: Is this something that worth a coremltools GitHub issue? Or do you think the coremltools behavior is better in 3.1? Overriding the class names sounds bad to me.

Comment (srikris) [reply src/python/turicreate/toolkits/sound_classifier/sound_classifier.py]: No I mis-understood the behavior. The new API is actually cleaner.

**Pull request merged**